### PR TITLE
CI: Make BPF test suite a first-class citizen

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

The BPF test suite has outgrown its current home in stable-perf

#### Summary of Changes

- Move BPF test suite to its own build step
- Dump BPF test asm listings, ix counts and shared objects to a build artifact to ease debugging

### CODE REVIEW

File: ci/test-stable.sh [hunk 1]
```diff
@@ -35,24 +31,51 @@ case $testName in
...
   _ "$cargo" stable test \
     --manifest-path programs/bpf/Cargo.toml \
-    --no-default-features --features=bpf_c,bpf_rust -- --nocapture
+    --no-default-features --features=bpf_c,bpf_rust --exclude assert_instruction_count -- --nocapture
 
+  # BPF program instruction count assertion
+  bpf_target_path=programs/bpf/target
+  _ "$cargo" stable test \
+    --manifest-path programs/bpf/Cargo.toml \
+    --no-default-features --features=bpf_rust assert_instruction_count -- --nocapture &> "${bpf_target_path}"/deploy/instuction_counts.txt
```
Comment (dmakarov) [inline ci/test-stable.sh:64]: It seems in this case assert_instruction_count will not execute C-based programs (only Rust-based ones), or do I miss something?  https://github.com/solana-labs/solana/blob/4e1333fbe67885dfb175844f5d499802aeac1772/programs/bpf/tests/programs.rs#L1277-L1292

Comment (t-nelson) [reply ci/test-stable.sh:64]: Whoops! Nice catch.  Copypasta'd this from my commit where we were debugging Rust `rand`

**Review by dmakarov — APPROVED** [review]
Great work!

**Comment by t-nelson** [comment]
> Great work!

Hmm... Cargo thinks otherwise. I wonder why I didn't hit this error locally :thinking:

**Comment by dmakarov** [comment]
> > Great work!
> 
> Hmm... Cargo thinks otherwise. I wonder why I didn't hit this error locally 🤔

In such matters cargo always has the last word :-)

**Pull request merged**